zerver/tests/test_subs.py

```diff
@@ -715,15 +768,31 @@ def test_rename_stream(self) -> None:
         self.assertNotIn(self.example_user('prospero').id,
                          notified_user_ids)
 
-    def test_rename_stream_requires_realm_admin(self) -> None:
+        # Test renaming of stream by stream admin.
+        do_change_user_role(user_profile, UserProfile.ROLE_MEMBER)
+        new_stream = self.make_stream('new_stream', realm=user_profile.realm)
+        self.subscribe(user_profile, 'new_stream')
+        sub = get_subscription('new_stream', user_profile)
+        do_change_subscription_property(user_profile, sub, new_stream, "role", Subscription.ROLE_STREAM_ADMINISTRATOR)
+        with tornado_redirected_to_list(events):
+            result = self.client_patch(f'/json/streams/{new_stream.id}',
+                                       {'new_name': ujson.dumps('stream_rename')})
+        self.assert_json_success(result)
+
+        stream_rename_exists = get_stream('stream_rename', realm)
+        self.assertTrue(stream_rename_exists)
```

Comment:
I don't understand this test -- we don't seem to inspect `events` after we're done?

Reply:
We should at least assert the number of events is as expected.